File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.IO;
using DevExpress.XtraPrinting;
using DevExpress.XtraRichEdit.Utils;
using DevExpress.XtraRichEdit.API.Native;
using DevExpress.Office.Utils;

namespace WindowsApplication111 {
    public partial class Form1 : Form {
        PrintableComponentLink link; 
        public Form1() {
            InitializeComponent();
            link = new PrintableComponentLink(new PrintingSystem());
            link.Component = this.richEditControl1;
            link.PrintingSystem.AfterMarginsChange += new MarginsChangeEventHandler(PrintingSystem_AfterMarginsChange);
            link.PrintingSystem.AfterBuildPages += PrintingSystem_AfterBuildPages;
            string path = Path.Combine(Application.StartupPath, @"..\..\Document.rtf");
            this.richEditControl1.LoadDocument(path);
        }

        void PrintingSystem_AfterBuildPages(object sender, EventArgs e)
        {
            link.PrintingSystem.SetCommandVisibility(PrintingSystemCommand.PageMargins, CommandVisibility.All);
        }

        #region #changemargins
        void PrintingSystem_AfterMarginsChange(object sender, MarginsChangeEventArgs e) {
            // Change document margins in the source RichEditControl
            SectionMargins margins = this.richEditControl1.Document.Sections[0].Margins;
            switch (e.Side) {
                case MarginSide.Left:
                    margins.Left = Units.HundredthsOfInchToDocuments((int)e.Value);
                    break;
                case MarginSide.Right:
                    margins.Right = Units.HundredthsOfInchToDocuments((int)e.Value);
                    break;
                case MarginSide.Top:
                    margins.Top = Units.HundredthsOfInchToDocuments((int)e.Value);
                    break;
                case MarginSide.Bottom:
                    margins.Bottom = Units.HundredthsOfInchToDocuments((int)e.Value);
                    break;
                default:
                    break;
            }
            
            link.CreateDocument();
        }
        #endregion #changemargins

        private void btnShowPrintPreview_Click(object sender, EventArgs e) {
            PrintTool tool = new PrintTool(link.PrintingSystem);
            link.CreateDocument();
            tool.ShowPreviewDialog();
        }

    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.IO
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraRichEdit.Utils
Imports DevExpress.XtraRichEdit.API.Native
Imports DevExpress.Office.Utils

Namespace WindowsApplication111
    Partial Public Class Form1
        Inherits Form

        Private link As PrintableComponentLink
        Public Sub New()
            InitializeComponent()
            link = New PrintableComponentLink(New PrintingSystem())
            link.Component = Me.richEditControl1
            AddHandler link.PrintingSystem.AfterMarginsChange, AddressOf PrintingSystem_AfterMarginsChange
            AddHandler link.PrintingSystem.AfterBuildPages, AddressOf PrintingSystem_AfterBuildPages
            Dim path As String = System.IO.Path.Combine(Application.StartupPath, "..\..\Document.rtf")
            Me.richEditControl1.LoadDocument(path)
        End Sub

        Private Sub PrintingSystem_AfterBuildPages(ByVal sender As Object, ByVal e As EventArgs)
            link.PrintingSystem.SetCommandVisibility(PrintingSystemCommand.PageMargins, CommandVisibility.All)
        End Sub

        #Region "#changemargins"
        Private Sub PrintingSystem_AfterMarginsChange(ByVal sender As Object, ByVal e As MarginsChangeEventArgs)
            ' Change document margins in the source RichEditControl
            Dim margins As SectionMargins = Me.richEditControl1.Document.Sections(0).Margins
            Select Case e.Side
                Case MarginSide.Left
                    margins.Left = Units.HundredthsOfInchToDocuments(CInt((e.Value)))
                Case MarginSide.Right
                    margins.Right = Units.HundredthsOfInchToDocuments(CInt((e.Value)))
                Case MarginSide.Top
                    margins.Top = Units.HundredthsOfInchToDocuments(CInt((e.Value)))
                Case MarginSide.Bottom
                    margins.Bottom = Units.HundredthsOfInchToDocuments(CInt((e.Value)))
                Case Else
            End Select

            link.CreateDocument()
        End Sub
        #End Region ' #changemargins

        Private Sub btnShowPrintPreview_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnShowPrintPreview.Click
            Dim tool As New PrintTool(link.PrintingSystem)
            link.CreateDocument()
            tool.ShowPreviewDialog()
        End Sub

    End Class
End Namespace